WalletWasabi/Blockchain/Transactions/AllTransactionStore.cs

```diff
@@ -14,8 +14,11 @@ namespace WalletWasabi.Blockchain.Transactions
 {
 	public class AllTransactionStore
 	{
-		public AllTransactionStore(Network network)
+		public AllTransactionStore(string workFolderPath, Network network)
 		{
+			WorkFolderPath = Guard.NotNullOrEmptyOrWhitespace(nameof(workFolderPath), workFolderPath, trim: true);
+			IoHelpers.EnsureDirectoryExists(WorkFolderPath);
```

Comment:
Ideally we shouldn't do this kind of staff in a constructor (I know we have constructors that do more than all the class methods together but given these PRs are for improving the current situation it would worth to start improving this too). Ctors should not create directories.

Reply:
I agree. I prefer not to change too many things at once. I would like to improve this in a future PR. I'll open a issue for that when/if this PR is merged.

Reply:
Why not put this into the InitializeAsync?